ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -277,10 +275,9 @@ data DiffusionArguments = DiffusionArguments {
     , daPeerSelectionTargets :: PeerSelectionTargets
       -- ^ selection targets for the peer governor
 
-    , daStaticLocalRootPeers :: [(Socket.SockAddr, PeerAdvertise)]
-    , daLocalRootPeers       :: [(DomainAddress, PeerAdvertise)]
-    , daPublicRootPeers      :: [DomainAddress]
-    , daUseLedgerAfter       :: UseLedgerAfter
+    , daLocalRootPeersVar    :: StrictTVar m [(Int, Map RelayAddress PeerAdvertise)]
+    , daPublicRootPeersVar   :: StrictTVar m [RelayAddress]
+    , daUseLedgerAfterVar    :: StrictTVar m UseLedgerAfter
```

Comment:
Do we write to all three `TVar`s? Or all some of them read only as far as network is concerned?

Reply:
This patch is more than updating `ouroboros-consensus`, as indicated in the git message.

Reply:
We do write to all `TVar`s when one triggers the hot config update via SIGHUP

Reply:
I will update the commit and make it more descriptive!

Reply:
But isn't that done in `cardano-node`?

Reply:
Yes, and the Diffusion Layer provides the TVars to be updated by `cardano-node`

Reply:
it's the other way  around, 'DiffusionArguments` pass data from `cardano-node` to `ouroboros-network`.  The `TVar`'s must be created in `cardano-node` and in `ouroboros-network` we probably only need to read them.

Reply:
Ah, I see. So what happens is, only `daLocalRootPeersVar` is written by `ouroboro-network` the others are read-only I believe, but can change if over at `cardano-node` someone triggers local config update (i.e. `cardano-node` writes to them).